src/genie/libs/parser/iosxe/show_cts.py

```diff
@@ -149,3 +148,138 @@ def filter_lines(raw_output, remove_lines):
                 continue
         return cts_pacs_dict
 
+
+
+# =================================
+# Schema for:
+#  * 'show cts role-based counters'
+# =================================
+class ShowCtsRoleBasedCountersSchema(MetaParser):
+    """Schema for show cts role-based counters."""
+
+    schema = {
+        "cts_rb_count": {
+            int: {
+                "src_group": Or(str, int),
+                "dst_group": Or(str, int),
+                "sw_denied_count": int,
+                "hw_denied_count": int,
+                "sw_permit_count": int,
+                "hw_permit_count": int,
+                "sw_monitor_count": int,
+                "hw_monitor_count": int
+            }
+        }
+    }
+
+
+# =================================
+# Parser for:
+#  * 'show cts role-based counters'
+# =================================
+class ShowCtsRoleBasedCounters(ShowCtsRoleBasedCountersSchema):
+    """Parser for show cts role-based counters"""
+
+    cli_command = 'show cts role-based counters'
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+
+        cts_rb_count_dict = {}
+        # Role-based IPv4 counters
+        # From    To      SW-Denied  HW-Denied  SW-Permitt HW-Permitt SW-Monitor HW-Monitor
```

Comment:
Don't need all 50 lines, 10 should be plenty. Once that is fixed, please put in PR